### PR TITLE
Adds AUMCdb Transformation Configuration

### PR DESCRIPTION
**Pull request by rvandewater**
Additionally, some fixes for the MIMIC-IV example

## Summary by CodeRabbit

- **New Features**
	- Introduced comprehensive README files for AUMC and MIMIC-IV examples, detailing dataset extraction and processing instructions.
	- Added configuration files to manage patient data and streamline data processing workflows.
	- Implemented scripts for efficient data handling and parallel processing capabilities.

- **Bug Fixes**
	- Clarified directory paths and variable usage in MIMIC-IV README to prevent confusion.

- **Documentation**
	- Enhanced README files with installation instructions, usage guidelines, and notes on potential memory issues.

- **Chores**
	- Added validation checks in scripts to improve robustness and user guidance.

### CODE REVIEW

**Comment by rvandewater** [comment]
The pipeline seems to be running without errors now @mmcdermott. Although it seems like the extract_AUMC.yaml file with _n_patients_per_shard_ does not result in multiple sharded files.

File: AUMC_Example/configs/extract_AUMC.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+input_dir: ${oc.env:AUMC_PRE_MEDS_DIR}
+cohort_dir: ${oc.env:AUMC_MEDS_COHORT_DIR}
+
+etl_metadata:
+  dataset_name: AUMCdb
+  dataset_version: 1.0.2
+
+stage_configs:
+  shard_events:
+    infer_schema_length: 999999999
```
Comment (mmcdermott) [inline AUMC_Example/configs/extract_AUMC.yaml:27]: Are your inputs in csv or parquet format? This is useful for CSVs, but if they are in csvs won't you need time formatting strings as well in the event configs yaml?

File: AUMC_Example/joint_script.sh
Comment (mmcdermott) [inline AUMC_Example/joint_script.sh:1]: You can delete this. We only need the runner script now, not these joint scripts.

File: AUMC_Example/pre_MEDS.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    offset_col: str | list[str],
+    pseudotime_col: str | list[str],
+    output_data_cols: list[str] | None = None,
+    warning_items: list[str] | None = None,
+) -> Callable[[pl.LazyFrame, pl.LazyFrame], pl.LazyFrame]:
+    """Returns a function that joins a dataframe to the `patient` table and adds pseudotimes.
+
+    Also raises specified warning strings via the logger for uncertain columns.
+
+    TODO
```
Comment (mmcdermott) [inline AUMC_Example/pre_MEDS.py:87]: You should fill this out

File: AUMC_Example/pre_MEDS.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+            *output_data_cols,
+        )
+
+    return fn
+
+
+@hydra.main(version_base=None, config_path="configs", config_name="pre_MEDS")
+def main(cfg: DictConfig):
+    """Performs pre-MEDS data wrangling for AUMCdb.
+
```
Comment (mmcdermott) [inline AUMC_Example/pre_MEDS.py:135]: you should fill this out or just eliminate the empty lines

**Review by mmcdermott — APPROVED** [review]

**Comment by mmcdermott** [comment]
Note that:
1. I didn't review the logic of the pre-meds step, because I'm not familiar with AUMCdb. 
2. There are some format and empty import suggestions that coderabbit has that you should accept. They have committable suggestions that can make it easy to follow them. 
3. You'll likely have to make some changes for the PR and tests check which will likely overlap with the coderabbit check -- I've just approved them so they are running now.

Also, flagging this PR generally for @kingrc15, who is working on cleaning up the eICU example as well.

File: AUMC_Example/README.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+so that the partition names are correct before use.**_ The memory and time costs are viable in the current
+configuration, but if your nodes are sufficiently different you may need to adjust those as well.
+
+The `local_parallelism_runner.yaml` file (downloaded above) runs each stage via separate processes on the
+launching machine. There are no additional arguments needed for this stage beyond the `N_WORKERS` environment
+variable and there is nothing to customize in this file.
+
+To profile the time and memory costs of your ETL, add the `do_profile=true` flag at the end.
+
+## Limitations / TO-DOs:
```
[the remark this answers is not in the document]
Comment (rvandewater) [reply AUMC_Example/README.md:82]: To be honest I just copied a lot from the MIMIC one as the underlying pipeline should be the same. I think this is the case for profiling as well?

File: AUMC_Example/configs/extract_AUMC.yaml (hunk 1 above)
Comment (rvandewater) [reply AUMC_Example/configs/extract_AUMC.yaml:27]: Inputs are csv

File: AUMC_Example/pre_MEDS.py (hunk 2 above)
Comment (rvandewater) [reply AUMC_Example/pre_MEDS.py:87]: @prockenschaub do you have feedback on this?

File: AUMC_Example/README.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+   expected format of the command.
+
+5. (Optional) Generate preliminary code statistics and merge to external metadata. This is not performed
+   currently in the `joint_script*.sh` scripts.
+
+## Limitations / TO-DOs:
+
+Currently, some tables are ignored, including:
+
+INSERT TABLES IGNORED HERE
```
[the remark this answers is not in the document]
Comment (rvandewater) [reply AUMC_Example/README.md:83]: @prockenschaub

File: MIMIC-IV_Example/configs/extract_MIMIC.yaml [hunk 6]
```diff
@@ -25,6 +25,8 @@ etl_metadata:
 stage_configs:
   shard_events:
     infer_schema_length: 999999999
+  split_and_shard_subjects:
+    n_patients_per_shard: 1000
```
Comment (prockenschaub) [inline MIMIC-IV_Example/configs/extract_MIMIC.yaml:29]: Is this required for everyone?

File: eICU_Example/README.md [hunk 7]
```diff
@@ -9,12 +9,13 @@ up from this one).
...
-- [ ] Implementing the pre-MEDS processing step.
-    - [ ] Identifying the pre-MEDS steps for eICU
-- [ ] Testing the pre-MEDS processing step on live eICU-CRD.
-    - [ ] Test that it runs at all.
-    - [ ] Test that the output is as expected.
+- [x] Implementing the pre-MEDS processing step.
+    - [x] Identifying the pre-MEDS steps for eICU
+- [x] Testing the pre-MEDS processing step on live eICU-CRD.
+    - [x] Test that it runs at all.
+    - [x] Test that the output is as expected.
```
Comment (prockenschaub) [inline eICU_Example/README.md:16]: Did you check these steps for eICU?

**Review by prockenschaub — CHANGES_REQUESTED** [review]

**Comment by rvandewater** [comment]
@mmcdermott Request is ready to merge from our side

File: AUMC_Example/configs/extract_AUMC.yaml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+stage_configs:
+  split_and_shard_subjects:
+    n_subjects_per_shard: 1000
+
+stages:
+  - shard_events
+  - split_and_shard_subjects
+  - convert_to_sharded_events
+  - merge_to_MEDS_cohort
+  - extract_code_metadata
```
Comment (mmcdermott) [inline AUMC_Example/configs/extract_AUMC.yaml:34]: As you don't have any code metadata blocks in your event configs, we can delete this stage (just deleting this line will work). It won't affect the later stages or anything and will just make things a bit cleaner.

File: AUMC_Example/pre_MEDS.py [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+    Args: 
+        table_name: name of the AUMCdb table that should be joined
+        offset_col: list of all columns that contain time offsets since the patient's first admission
+        pseudotime_col: list of all timestamp columns derived from `offset_col` and the linked `patient` 
+            table
+        output_data_cols: list of all data columns included in the output 
+        warning_items: any warnings noted in the table_preprocessors.yaml
+
+    Example:
```
Comment (mmcdermott) [inline AUMC_Example/pre_MEDS.py:91]: nitpick: For an `Example` block, I'd recommend including a doctest style example that way it shows actual usage and can serve as a test of the logic.

File: AUMC_Example/pre_MEDS.py [hunk 10]
```diff
@@ -0,0 +1,237 @@
...
+from loguru import logger
+from omegaconf import DictConfig, OmegaConf
+
+from MEDS_transforms.utils import get_shard_prefix, hydra_loguru_init, write_lazyframe
+
+ADMISSION_ID = "admissionid"
+PATIENT_ID = "patientid"
+
+def load_raw_aumc_file(fp: Path, **kwargs) -> pl.LazyFrame:
+    """Load a raw AUMCdb file into a Polars DataFrame.
```
Comment (mmcdermott) [inline AUMC_Example/pre_MEDS.py:23]: nitpick: For a function this simple, a one-line docstring is reasonable too, imo.

**Review by mmcdermott — APPROVED** [review]

**Comment by mmcdermott** [comment]
@rvandewater and @prockenschaub -- if you, in your terminal where you're running `git commit` on these files, install `pre-commit` via `pip install -e .[dev]` then install the pre-commit hooks via `pre-commit install` it should automatically check and correct the issues in the PR check for you on every commit.

File: AUMC_Example/pre_MEDS.py (hunk 9 above)
Comment (rvandewater) [reply AUMC_Example/pre_MEDS.py]: Added example logic

File: AUMC_Example/pre_MEDS.py (hunk 10 above)
Comment (rvandewater) [reply AUMC_Example/pre_MEDS.py:24]: Added example logic

**Pull request merged**